lxd/db/storage_volumes.go

```diff
@@ -690,10 +690,7 @@ SELECT storage_volumes_snapshots.name FROM storage_volumes_snapshots
    AND storage_pools.name=?
    AND (storage_volumes.node_id=? OR storage_volumes.node_id IS NULL AND storage_pools.driver IN ` + query.Params(len(remoteDrivers)) + `)`
```

Comment:
Dunno why there is no `ORDER BY` clause here like there is in `GetNextInstanceSnapshotIndex()`. Anyway, here too I'd expect an `ORDER BY ... DESC LIMIT 1` to only pull the name of the latest volume snapshot from the DB rather then all of them.